### ISSUE TITLE
Cached progress disappears on subtitle change

### ISSUE THREAD

**Issue by smhmd** [mpv-player/mpv#7780]
<details>

<summary>Important Information</summary>

- mpv version `0.32.0`
- Linux Distribution and Version `Linux manjaro 5.6.12-1-MANJARO`
- Source of the mpv binary [`Arch repository`]	(https://www.archlinux.org/packages/community/x86_64/mpv/)
- If known which version of mpv introduced the problem `n/a`
- Window Manager and version `dwm-6.2`
- GPU driver and version `Intel Corporation 4th Gen Core Processor Integrated Graphics Controller (rev 06)`
- Possible screenshot or video of visual glitches `n/a`

</details>

### Reproduction steps
- command: `mpv 'http://example.com/file.mkv' --no-config --log-file=output.txt`
- The file is an HTTP link of type MKV
- When the file opens, the subtitle track is set to none. When I enable the embedded subtitle, the cached progress evaporates and the video starts loading the once cached part.
- applicable with a `--playlist=file --prefetch-playlist`

### Expected behavior
Subs get enabled; cached progress remains. 

### Actual behavior
Subs get enabled; cached progress disappears.
 
### Log file
https://0x0.st/if3M.txt

**Comment by wm4** [comment]
That is normal and expected. If a track is not selected, the cache won't contain any data for that track. So the file needs to be read again if the track is selected later.

In your case you could just get the subtitle selected by default, with --sid=1 or --slang=eng.

In theory we could always demux and subtitle tracks, and just pretend they're selected/deselected when the user does so. But not sure if that wouldn't conflict with certain use-cases.

**Comment by smhmd** [comment]
Thanks for the workaround.

**Comment by tamilselvan0X0** [comment]
> That is normal and expected. If a track is not selected, the cache won't contain any data for that track. So the file needs to be read again if the track is selected later.

Please no, this will not be a problem when the file is in local, in fact it will just be the better choice in the trade-off between memory and disk read, but for the file in network it is bad, even worse for the poor network connection.

Maybe the network cache could be stored in the disk and switching audio/subtitle could make it load from there?

**Comment by rexackermann** [comment]
> > That is normal and expected. If a track is not selected, the cache won't contain any data for that track. So the file needs to be read again if the track is selected later.
> 
> Please no, this will not be a problem when the file is in local, in fact it will just be the better choice in the trade-off between memory and disk read, but for the file in network it is bad, even worse for the poor network connection.
> 
> Maybe the network cache could be stored in the disk and switching audio/subtitle could make it load from there?

Yes, I also think this would be a good idea. On slower networks it's really annoying when it re-caches again
